com.microsoft.mrtk.input/Visualizers/ControllerVisualizer.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.OpenXR;
+using UnityEngine;
+using UnityEngine.InputSystem.Utilities;
+using UnityEngine.XR;
+using UnityEngine.XR.Interaction.Toolkit;
+using UnityInputSystem = UnityEngine.InputSystem;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Basic controller visualizer which draws a generic motion controller when one is detected
+    /// </summary>
+    [AddComponentMenu("MRTK/Input/Controller Visualizer")]
+    public class ControllerVisualizer : MonoBehaviour
+    {
+        [SerializeField]
+        [Tooltip("The XRNode on which this hand is located.")]
+        private XRNode handNode = XRNode.LeftHand;
+
+        /// <summary> The XRNode on which this hand is located. </summary>
+        public XRNode HandNode { get => handNode; set => handNode = value; }
+
+        [SerializeField]
+        [Tooltip("A fallback controller model to render in case the platform model fails to load")]
+        private GameObject fallbackControllerModel;
+
+        // caching the controller we belong to
+        private XRBaseController xrController;
+
+        // caching the controller model provider we are using we belong to
+        private ControllerModel controllerModelProvider;
+
+        // The controller characteristics we want the input device to have;
+        private InternedString targetUsage;
+
+        /// <inheritdoc />
+        protected bool isControllerTracked => xrController.currentControllerState.inputTrackingState.HasPositionAndRotation();
+
+        protected void OnEnable()
+        {
+            Debug.Assert(handNode == XRNode.LeftHand || handNode == XRNode.RightHand, $"HandVisualizer has an invalid XRNode ({handNode})!");
+
+            ControllerLookup[] lookups = GameObject.FindObjectsOfType(typeof(ControllerLookup)) as ControllerLookup[];
```

Comment:
You should probably search recursively from the interactionmanager or the rig, rather than searching the entire scene. I believe that's how we do it in the other instances where we grab a reference to the `ControllerLookup`? I could be wrong, but that's what my gut tells me

Reply:
It looks like the other instance where we use `ControllerLookup` is the Solver class uses the same thing we do here. I think we should rethink how ControllerLookup should be done in general, but I think it is out of scope for this specific PR.